var/spack/repos/builtin/packages/r-tidyverse/package.py

```diff
@@ -16,60 +16,97 @@ class RTidyverse(RPackage):
     <https://tidyverse.org>."""
 
     homepage = "https://tidyverse.tidyverse.org/"
-    url      = "https://cloud.r-project.org/src/contrib/tidyverse_1.2.1.tar.gz"
-    list_url = "https://cloud.r-project.org/src/contrib/Archive/tidyverse"
+    cran = "tidyverse"
 
+    version('1.3.1', sha256='83cf95109d4606236274f5a8ec2693855bf75d3a1b3bc1ab4426dcc275ed6632')
     version('1.3.0', sha256='6d8acb81e994f9bef5e4dcf908bcea3786d108adcf982628235b6c8c80f6fe09')
     version('1.2.1', sha256='ad67a27bb4e89417a15338fe1a40251a7b5dedba60e9b72637963d3de574c37b')
 
     depends_on('r+X', type=('build', 'run'))
-    depends_on('r@3.2:', when='@1.3.0:', type=('build', 'run'))
+
+    with when('@1.3.1:'):
+        depends_on('r@3.3:', type=('build', 'run'))
+        depends_on('r-broom@0.7.6:', type=('build', 'run'))
+        depends_on('r-cli@2.4.0:', type=('build', 'run'))
+        depends_on('r-crayon@1.4.1:', type=('build', 'run'))
+        depends_on('r-dbplyr@2.1.1:', type=('build', 'run'))
+        depends_on('r-dplyr@1.0.5:', type=('build', 'run'))
+        depends_on('r-dtplyr@1.1.0:', type=('build', 'run'))
+        depends_on('r-forcats@0.5.1:', type=('build', 'run'))
+        depends_on('r-googledrive@1.0.1:', type=('build', 'run'))
+        depends_on('r-googlesheets4@0.3.0:', type=('build', 'run'))
+        depends_on('r-ggplot2@3.3.3:', type=('build', 'run'))
+        depends_on('r-haven@2.3.1:', type=('build', 'run'))
+        depends_on('r-hms@1.0.0:', type=('build', 'run'))
+        depends_on('r-httr@1.4.2:', type=('build', 'run'))
+        depends_on('r-jsonlite@1.7.2:', type=('build', 'run'))
+        depends_on('r-lubridate@1.7.10:', type=('build', 'run'))
+        depends_on('r-magrittr@2.0.1:', type=('build', 'run'))
+        depends_on('r-modelr@0.1.8:', type=('build', 'run'))
+        depends_on('r-pillar@1.6.0:', type=('build', 'run'))
+        depends_on('r-purrr@0.3.4:', type=('build', 'run'))
+        depends_on('r-readr@1.4.0:', type=('build', 'run'))
+        # r-readxl@1.3.1: see @1.3.0:
```

Comment:
While I understand the motivation for using `with when()` this structure makes reviewing r packages more difficult. It probably also makes them more difficult to write. For example, I have to scroll down to find the `r-readxl` dependency. Having the constraints of each dependency grouped together, and in the order that they appear in the DESCRIPTION file make things easier to write and review, IMO. Could you please redo this in the more conventional/historical way? Thanks.

Reply:
Thanks for your thoughts on this. I actually started with having the dependencies grouped together but it made it really hard to read. For other packages, where only a few dependency change when a new release comes out, this might be different and I am with you on this but this packages is kind of special because the developers seem to change nearly all dependency versions for a new release. Grouping it together by version made writing and also checking it so much easier for me here.